trin-core/src/portalnet/overlay_service.rs

```diff
@@ -79,6 +79,20 @@ const EXPECTED_NON_EMPTY_BUCKETS: usize = 17;
 /// Bucket refresh lookup interval in seconds
 const BUCKET_REFRESH_INTERVAL_SECS: u64 = 60;
 
+/// An action that the overlay may perform.
+#[derive(Debug)]
+pub enum OverlayCommand<TContentKey> {
+    /// Send a request through the overlay.
+    Request(OverlayRequest),
+    /// Perform a find content query through the overlay.
+    FindContentQuery {
```

Comment:
> Here, I can't think of a better name than "command" so I'm going to leave this as is for now, unless you have a suggestion.

Yeah, I don't have a better suggestion either, so no objection to leaving it as command. But I highly recommend the following suggestion:

> Explicitly defining the difference between a "command" and "request" and how they are used would be sufficient.

The comments in 82 and 85 could use some love. How does a request differ from a command here, semantically? Is it that a Command is any kind of action we might ask the overlay service loop to perform (which might include multiple back-and-forth network requests), and a Request always has a single network request that it maps to 1:1?

If I have that wrong and it's hard to explain even with a couple sentences, that could be an important signal that we haven't found the right architectural choice yet.

Reply:
Agreed. You've described the basic idea that I had in mind.